pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,490 @@
+
+\lstset{style=mystyle}
+
+\newcommand{\MultiSets}{\mathrm{MultiSets}}
+\newcommand{\len}{\mathrm{len}}
+\newcommand{\din}{d_{in}}
+\newcommand{\dout}{d_{out}}
+\newcommand{\Relation}{\mathrm{Relation}}
+\newcommand{\X}{\mathcal{X}}
+\newcommand{\Y}{\mathcal{Y}}
+\newcommand{\True}{\texttt{True}}
+\newcommand{\False}{\texttt{False}}
+\newcommand{\clamp}{\texttt{clamp}}
+\newcommand{\horizline}{\noindent\rule{\textwidth}{1pt}}
+
+
+\newcommand{\silvia}[1]{{ {\color{blue}{(silvia)~#1}}}}
+\newcommand{\grace}[1]{{ {\color{purple}{(grace)~#1}}}}
+\newcommand{\connor}[1]{{ {\color{teal}{(connor)~#1}}}}
+\newcommand{\sfi}{\text{\textcolor{olive}{(Salil's feedback incorporated.)}}}
+\newcommand{\todosfi}{\todo{Add Salil's feedback.}}
+
+\newcommand{\metricDefn}[1]{The definition of \emph{#1} in the ``proof definitions'' document tells how the distance between data is calculated.}
+
+\newcommand{\inRust}[2]{See \url{#2}.}
+\newcommand{\inOpenDPRust}[2]{In OpenDP (Rust), this is called \texttt{#1}. See \url{#2}.}
+\newcommand{\inPython}[2]{In Python, this is called \texttt{#1}. See \url{#2}.}
+\newcommand{\questionc}[1]{\textcolor{red}{\textbf{Question:} #1}}
+\newcommand{\questionr}[1]{\textcolor{blue}{\textbf{Question for reviewers:}}\textcolor{red}{~#1}}
+\newcommand{\T}{\texttt{T}}
+\newcommand{\data}{\texttt{data}}
+\newcommand{\todo}{{\textcolor{red}{TODO: }}}
+\newcommand{\todonei}{{\textcolor{red}{TODO (future -- not enough info yet): }}}
+\newcommand{\iffText}{\text{if and only if}}
+
+
+\title{List of definitions used in the pseudocode}
+\author{S\'ilvia Casacuberta, Grace Tian, and Connor Wagaman}
+\date{June--July 2021}
+
+\begin{document}
+
+\maketitle
+
+We use the following guideline: if a term appears in the preconditions \& pseudocode section, then this term is defined in the ``List of definitions used in the pseudocode" document. Otherwise, it appears in the ``List of definitions used in the proofs" document. 
+
+In both cases, we maintain the terms in alphabetical order within each section. ``TODOs'' should be included at the end of the corresponding section. On the other hand, ``TODOs'' which better specify an already-defined term should be included immediately following the definition of that term. Examples should never be part of the definition, but we encourage their use right after the definition of a term.
+
+We also recommend linking to the Rust Standard Library in the cases where the term is already defined there.
+
+\tableofcontents
+
+\section{Types}
+% any types should be listed here
+
+\begin{definition}[\texttt{bool}]
+    The type \texttt{bool} represents a value, which can only be either \texttt{True} or \texttt{False}. If you \texttt{cast} a \texttt{bool} into an integer, \texttt{True} will be \texttt{1} and \texttt{False} will be \texttt{0}.
+\end{definition}
+
+\begin{definition}[\texttt{::Carrier}]
+\texttt{SomeDomain::Carrier} is the type of a member in \texttt{SomeDomain}, where \texttt{SomeDomain} is a domain. 
```

Comment:
I'd move this to the Domains section.  It doesn't make sense yet, since the concept of Domains hasn't been defined.

Reply:
I'm referring to the def of ::Carrier

Reply:
structure issue -- we should discuss how this document should be structured. See [this reply](https://github.com/opendp/whitepapers/pull/4#discussion_r671339240) for more discussion of why alphabetization was used.

Reply:
Cross-referencing would help - so the reader knows where to find the definition of Domain.  (Right now looking at the def of Carrier, it's not clear that Domain is something that is defined elsewhere in the document.)

Reply:
Done here